src/main/java/gregtech/api/capability/impl/FuelRecipeLogic.java

```diff
@@ -192,19 +241,28 @@ private int tryAcquireNewRecipe(FluidStack fluidStack) {
                 this.previousRecipe = currentRecipe;
             }
         }
+
+        //TODO, should this be done here, as it is called in the middle of a loop over valid fluid tanks?
+        this.invalidInputsForRecipes = (currentRecipe == null);
```

Comment:
this should be set after here, unless we refactor this to look more like the logic in AbstractRecipeLogic, where the result of findRecipe is already looped through the tank list